### PR TITLE
PIM-3759: add behat scenario

### PR DESCRIPTION
**Pull request by skeleton**
| Q                    | A
| -------------------- | ---
| Bug fix?             | N
| New feature?         | N
| BC breaks?           | N
| CI currently passes? |Y
| Tests pass?          | Y
| Scenarios pass?      | ?
| Checkstyle issues?*  | N
| PMD issues?**        | N
| Changelog updated?   | N
| Fixed tickets        | PIM-3759
| DB schema updated?   | N
| Migration script?    | N
| Doc PR               |

### CODE REVIEW

File: features/product/filtering/filter_products_per_completeness.feature [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    And I should see products BOOTWXS, BOOTBS, BOOTBL, BOOTRXS
+
+  Scenario: Successfully filter complete productsfor default tablet channel
+    Given I am logged in as "Mary"
+    And I am on the products page
+    And I filter by "Complete" with value "yes"
+    Then the grid should contain 0 elements
+    And I filter by "Channel" with value "Mobile"
+    Then the grid should contain 1 elements
+    And I should see products BOOTBXS
```
Comment (sumbobyboys) [inline features/product/filtering/filter_products_per_completeness.feature]: please configure your code editor to add a blank line at the end of file

File: features/product/filtering/filter_products_per_completeness.feature [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+  Scenario: Successfully filter complete products for default mobile channel
+    Given I am logged in as "Peter"
+    And I am on the products page
+    And I filter by "Complete" with value "yes"
+    Then the grid should contain 1 elements
+    And I should see products BOOTBXS
+
+  Scenario: Successfully filter uncomplete products for default tablet channel
+    Given I am logged in as "Mary"
```
Comment (sumbobyboys) [inline features/product/filtering/filter_products_per_completeness.feature]: as discussed together you should use the same user and switch scope

Comment (sumbobyboys) [inline features/product/filtering/filter_products_per_completeness.feature]: after that you can move the login and the "got to product page" to the background section

**Comment by sumbobyboys** [comment]
Nice PR !
Please remember to add the content of CONTRIBUTING.md to your PR description and run it on the CI :)

File: features/product/filtering/filter_products_per_completeness.feature [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    And I am on the products page
+    And I filter by "Complete" with value "no"
+    Then the grid should contain 5 elements
+    And I should see products BOOTBXS, BOOTWXS, BOOTBS, BOOTBL, BOOTRXS
+    And I filter by "Channel" with value "Mobile"
+    Then the grid should contain 4 elements
+    And I should see products BOOTWXS, BOOTBS, BOOTBL, BOOTRXS
+
+  Scenario: Successfully filter complete productsfor default tablet channel
```
Comment (nidup) [inline features/product/filtering/filter_products_per_completeness.feature]: typo productsfor -> products for

File: features/Context/DataGridContext.php [hunk 4]
```diff
@@ -104,6 +104,18 @@ public function iFilterByPriceWithEmptyValue($filterName, $currency)
     }
 
     /**
+     * @param string $filterName
+     * @param string $currency
```
Comment (filipsalpe) [inline features/Context/DataGridContext.php]: currency?

File: features/Context/DataGridContext.php [hunk 5]
```diff
@@ -104,6 +104,18 @@ public function iFilterByPriceWithEmptyValue($filterName, $currency)
...
+     *
+     * @Then /^I filter by "([^"]*)" with value "(>|>=|=|<|<=|empty) (\d+[.]?\d*)"$/
```
Comment (filipsalpe) [inline features/Context/DataGridContext.php]: something wrong with this regex in the 'empty' case

File: features/product/filtering/filter_products_per_with_multiple_number_fields.feature [hunk 6]
```diff
@@ -33,35 +33,40 @@ Feature: Filter products
...
+  Scenario: Successfully filter products with the sames attributes
+    Given I filter by "Supplier" with value "03"
+    And I should be able to use the following filters:
+      | filter    | value | result                 |
+      | Component | empty | POST-1, POST-2         |
+      | Component | > 16  |                        |
+      | Component | < 16  |                        |
+      | Component | > 15  | MUG-2, MUG-3 and MUG-4 |
+      | Component | < 17  | MUG-2, MUG-3 and MUG-4 |
+      | Component | >= 16 | MUG-2, MUG-3 and MUG-4 |
```
Comment (sumbobyboys) [inline features/product/filtering/filter_products_per_with_multiple_number_fields.feature]: we should try to filter on more discriminant values

File: features/product/filtering/filter_products_per_with_multiple_prices.feature [hunk 7]
```diff
@@ -33,35 +33,46 @@ Feature: Filter products
...
+  Scenario: Successfully filter products with the sames attributes
+    Given I filter by "Transport" with value "= 15 EUR"
+    And I should be able to use the following filters:
+      | filter | value       | result                 |
+      | Margin | > 7 EUR     |                        |
+      | Margin | >= 7.01 EUR |                        |
+      | Margin | < 7 EUR     |                        |
+      | Margin | <= 6.99 EUR |                        |
+      | Margin | > 6 EUR     | MUG-2, MUG-3 and MUG-4 |
+      | Margin | > 6.99 EUR  | MUG-2, MUG-3 and MUG-4 |
```
Comment (sumbobyboys) [inline features/product/filtering/filter_products_per_with_multiple_prices.feature]: same here

**Pull request merged**